Added Specify Docker Build Sources section to Extending Docker Page (#6408)
magento/devdocs: src/cloud/docker/docker-extend.md

   docker-compose -f docker-compose.yml -f docker-compose-dev.yml run deploy bash
   ```

## Specify Docker Build Sources
To be able to locally test changes to images or if there is a requirement to make more extensive changes to the containers you will need to build them from source. This is accomplished by adding build context to the `docker-compose.override.yml`.

```yaml
version: '2.1'
services:
 web:
   build:
     context: ./vendor/magento/magento-cloud-docker/images/nginx/1.9/
```

> This example defines the build context for the web container, however this can be done with any of the images in `vendor/magento/magento-cloud-docker` or any image, including locally resourced outside of the project.

Reviewer: The suggestion above moves this sentence suggest to the paragraph introducing the example.

Wasn't sure what you mean by "locally resourced outside the project"--is there a clearer way to say that.  I think you mean if you have a Docker image that you've customized locally, you can also build it from source?

> When using this to make changes that require testing you can refresh the build of the container by running the following command:

Reviewer: ```suggestion
If you want to update the container configuration and test iteratively, you can run the following command to refresh the container build.
```

   docker-compose -f docker-compose.yml -f docker-compose-dev.yml run deploy bash
   ```

> ## Specify Docker Build Sources

Reviewer: ```suggestion
## Specify Docker build sources
```

To locally test changes to images or make more extensive changes to the containers, you must build them from source.

The following example shows how to build from source by adding `build:context` configuration  to the `docker-compose.override.yml` file. This example defines the build context for the web container. You can use the same technique to build from any of the images in `vendor/magento/magento-cloud-docker` or any other Docker image, including images locally resourced outside the project.

```yaml
version: '2.1'
services:
 web:
   build:
     context: ./vendor/magento/magento-cloud-docker/images/nginx/1.9/
```

> This example defines the build context for the web container, however this can be done with any of the images in `vendor/magento/magento-cloud-docker` or any image, including locally resourced outside of the project.

Reviewer: ```suggestion
```

> If you want to update the container configuration and test iteratively, you can run the following command to refresh the container build.

Reviewer: ```suggestion
If you want to update the container configuration and test iteratively, run the following command to refresh the container build.
```
